docs/serverless/functions.md

```diff
@@ -0,0 +1,175 @@
+---
+title: .NET Aspire Azure Functions integration (Preview)
+description: Learn how to integrate Azure Functions with .NET Aspire.
+ms.date: 10/30/2024
+zone_pivot_groups: dev-environment
+---
+
+# .NET Aspire Azure Functions integration (Preview)
+
+[!INCLUDE [includes-hosting](../includes/includes-hosting.md)]
+
+> [!IMPORTANT]
+> The .NET Aspire Azure Functions integration is currently in preview and is subject to change.
+
+[Azure Functions](/azure/azure-functions/functions-overview) is a serverless solution that allows you to write less code, maintain less infrastructure, and save on costs. The .NET Aspire Azure Functions integration enables you to develop, debug, and orchestrate an Azure Functions .NET project as part of the app host.
+
+It's expected that you've installed the required Azure tooling:
+
+:::zone pivot="visual-studio"
+
+- [Configure Visual Studio for Azure development with .NET](/dotnet/azure/configure-visual-studio)
+
+:::zone-end
+:::zone pivot="vscode"
+
+- [Configure Visual Studio Code for Azure development with .NET](/dotnet/azure/configure-vs-code)
+
+:::zone-end
+:::zone pivot="dotnet-cli"
+
+- [Install the Azure Functions Core Tools](/azure/azure-functions/functions-run-local?tabs=isolated-process&pivots=programming-language-csharp#install-the-azure-functions-core-tools)
+
+:::zone-end
+
+## Supported scenarios
+
+The .NET Aspire Azure Functions integration has several key supported scenarios. This section outlines the scenarios and provides details related to the implementation of each approach.
+
+### Supported triggers
+
+The following table lists the supported triggers for Azure Functions in the .NET Aspire integration:
+
+| Trigger | Attribute | Details |
+|--|--|--|
+| Azure Event Hubs trigger | `EventHubTrigger` | [📦 Aspire.Hosting.Azure.EventHubs](https://www.nuget.org/packages/Aspire.Hosting.Azure.EventHubs) |
+| Azure Service Bus trigger | `ServiceBusTrigger` | [📦 Aspire.Hosting.Azure.ServiceBus](https://www.nuget.org/packages/Aspire.Hosting.Azure.ServiceBus) |
+| Azure Storage Blobs trigger | `BlobTrigger` | [📦 Aspire.Hosting.Azure.Storage](https://www.nuget.org/packages/Aspire.Hosting.Azure.Storage) |
+| Azure Storage Queues trigger | `QueueTrigger` | [📦 Aspire.Hosting.Azure.Storage](https://www.nuget.org/packages/Aspire.Hosting.Azure.Storage) |
+| HTTP trigger | `HttpTrigger` | Supported without any additional resource dependencies. |
+| Timer trigger | `TimerTrigger` | [📦 Aspire.Hosting.Azure.Storage](https://www.nuget.org/packages/Aspire.Hosting.Azure.Storage) |
+
+> [!IMPORTANT]
+> Other Azure Functions trigger types aren't currently supported in the .NET Aspire Azure Functions integration.
```

Comment:
```suggestion
> Other Azure Functions triggers and bindings aren't currently supported in the .NET Aspire Azure Functions integration.
```

Input and output bindings also fall into the category of things we don't support.